tests/base/testSetFunctions.cpp

```diff
@@ -0,0 +1,326 @@
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+/**
+ * @file testCubicalComplex.cpp
+ * @ingroup Tests
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2015/08/28
+ *
+ * Functions for testing class CubicalComplex.
+ *
+ * This file is part of the DGtal library.
+ */
+
+///////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+#include <vector>
+#include <set>
+#include <random>
+#include <unordered_set>
+#include "DGtal/base/Common.h"
+#include "DGtal/base/SetFunctions.h"
+#include "DGtalCatch.h"
+///////////////////////////////////////////////////////////////////////////////
+
+using namespace DGtal;
+using namespace DGtal::functions;
+using namespace DGtal::functions::setops;
+
+////////////////////////////// unit tests /////////////////////////////////
+TEMPLATE_TEST_CASE_4( "SetFunctions module unit tests", "[set_functions]",
+                      Container,
+                      std::vector<int>,
+                      std::list<int>, 
+                      std::set<int>, 
+                      std::unordered_set<int> )
+
+{
+  int S1[ 10 ] = { 4, 15, 20, 17, 9, 7, 13, 12, 1, 3 }; 
+  int S2[ 6 ]  = { 17, 14, 19, 2, 3, 4 };
+  Container C1( S1, S1 + 10 );
+  Container C2( S2, S2 + 6 );
+  Container C1_minus_C2 = C1 - C2;
+  REQUIRE( C1_minus_C2.size() == 7 );
+  Container C2_minus_C1 = C2 - C1;
+  REQUIRE( C2_minus_C1.size() == 3 );
+  REQUIRE( ( C1_minus_C2 - C1 ).size() == 0 );
+  REQUIRE( ( C2_minus_C1 - C2 ).size() == 0 );
+  REQUIRE( ( C1_minus_C2 - C2 ).size() == C1_minus_C2.size() );
+  REQUIRE( ( C2_minus_C1 - C1 ).size() == C2_minus_C1.size() );
+  Container C1_union_C2 = C1 | C2;
+  Container C2_union_C1 = C2 | C1;
+  REQUIRE( C1_union_C2.size() == 13 );
+  REQUIRE( C1_union_C2.size() == C2_union_C1.size() );
+  REQUIRE( ( C1_minus_C2 | C2 ).size() == (C2_minus_C1 | C1 ).size() );
+
+  Container C1_intersection_C2 = C1 & C2;
+  Container C2_intersection_C1 = C2 & C1;
+  REQUIRE( C1_intersection_C2.size() == 3 );
+  REQUIRE( C1_intersection_C2.size() == C2_intersection_C1.size() );
+
+  REQUIRE( ( C1_minus_C2 | C1_intersection_C2 | C2_minus_C1 ).size() == C1_union_C2.size() );
+
+  Container C1_symdiff_C2 = C1 ^ C2;
+  Container C2_symdiff_C1 = C2 ^ C1;
+  REQUIRE( C1_symdiff_C2.size() == C2_symdiff_C1.size() );
+  REQUIRE( C1_symdiff_C2.size() == ( C1_union_C2 - C1_intersection_C2 ).size() );
+  REQUIRE( C1_symdiff_C2.size() == ( C1_minus_C2 | C2_minus_C1 ).size() );
+
+  // JOL: catch relation operator overloading interacts too much with
+  //      our own set operator overloading. We call the exact
+  //      operations.
+  // std::cout << "C1 D C2 = ";
```

Comment:
maybe you can cleanup this comment since you have removed the overloading on "=="